src/App.tsx

```diff
@@ -1,25 +1,89 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
-
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getPrepareTodos } from './components/helpers/api';
 import './App.scss';
-import Start from './components/Start';
-import { Finish } from './components/Finish';
+import TodoList from './components/TodoList';
+import Loading from './components/Loading';
+import ButtonsSort from './components/ButtonsSort';
 
-import { isLoading, getMessage } from './store';
+import {
+  startLoading,
+  finishLoading,
+  handleSuccess,
+  handleError,
+  getIsLoading,
+  getFinishLoading,
+  getError,
+  getVisibleTodos,
+} from './store';
 
 
 const App = () => {
-  const loading = useSelector(isLoading);
-  const message = useSelector(getMessage) || 'Ready!';
+  const dispatch = useDispatch();
+
+  const hasError = useSelector(getError);
+  const isLoading = useSelector(getIsLoading);
+  const todos = useSelector(getVisibleTodos);
+  const loaded = useSelector(getFinishLoading);
+  const [isToggle, setIsToggle] = useState(false);
```

Comment:
toggle is for switching something, if you want to describe visibility, then use something like `isVisible` or `isOpened`